lib/features/children/details/cubit/child_details_cubit.dart

```diff
@@ -105,6 +105,49 @@ class ChildDetailsCubit extends Cubit<ChildDetailsState> {
     }
   }
 
+  Future<void> cancelAllowance() async {
+    unawaited(
+      AnalyticsHelper.logEvent(
+        eventName: AmplitudeEvents.cancelRGA,
+        eventProperties: {
+          'child_name': _profile.firstName,
+        },
+      ),
+    );
+
+    try {
+      _emitLoading();
+      final isSuccess = await _editChildRepository.cancelAllowance(_profile.id);
+      if (isSuccess) {
+        _emitData();
+        emit(const ChildCancelAllowanceSuccessState());
+        //has to be awaited in order for the ui to update properly
+        await fetchChildDetails();
+      } else {
+        emit(
```

Comment:
I think that if isSuccess is now false  you'll show the dialog but the screen will stay stuck in the loading state, i think you need the _emitData there as well.